androidApp/src/androidMain/kotlin/at/asitplus/wallet/app/android/NdefDeviceEngagementService.kt

```diff
@@ -0,0 +1,300 @@
+        settings: TransferSettings,
+        encodedDeviceEngagement: ByteString,
+        handover: DataItem,
+        eDeviceKey: EcPrivateKey,
+        engagementDuration: Duration,
+    ) {
+        presentationStateModel.presentmentScope.launch {
+            presentationStateModel.state.first { it != PresentationStateModel.State.IDLE && it != PresentationStateModel.State.NO_PERMISSION && it != PresentationStateModel.State.CHECK_PERMISSIONS }
+            val transport = connectionMethods.advertiseAndWait(
+                role = MdocRole.MDOC,
+                transportFactory = MdocTransportFactory.Default,
+                options = MdocTransportOptions(
+                    bleUseL2CAP = settings.readerBleL2CapEnabled
+                ),
+                eSenderKey = eDeviceKey.publicKey,
+                onConnectionMethodsReady = {}
+            )
+            presentationStateModel.setMechanism(
+                MdocPresentmentMechanism(
+                    transport = transport,
+                    ephemeralDeviceKey = eDeviceKey,
+                    encodedDeviceEngagement = encodedDeviceEngagement,
+                    handover = handover,
+                    engagementDuration = engagementDuration,
+                    allowMultipleRequests = settings.presentmentAllowMultipleRequests
+                )
+            )
+            disableEngagementJob?.cancel()
+            disableEngagementJob = null
+            listenForCancellationFromUiJob?.cancel()
+            listenForCancellationFromUiJob = null
+        }
+    }
+
+    private suspend fun processCommandApdu(commandApdu: CommandApdu): ResponseApdu? {
+        Napier.i("NdefDeviceEngagementService: processCommandApdu")
+
+        if (!started) {
+            started = true
+            startEngagement()
+        }
+
+        try {
+            engagement?.let {
+                val responseApdu = it.processApdu(commandApdu)
+                return responseApdu
+            }
+        } catch (e: Throwable) {
+            Napier.e("NdefDeviceEngagementService: processCommandApdu", e)
+            e.printStackTrace()
+        }
+        return null
+    }
+
+    // Called by OS when an APDU arrives
+    override fun processCommandApdu(encodedCommandApdu: ByteArray, extras: Bundle?): ByteArray? {
+        // Bounce the APDU to processCommandApdu() above via the coroutine in I/O thread set up in onCreate()
+        commandApduChannel.trySend(CommandApdu.decode(encodedCommandApdu))
+        return null
+    }
+
+    override fun onDeactivated(reason: Int) {
+        Napier.i("NdefDeviceEngagementService: onDeactivated: reason=$reason")
+        started = false
+        // If the reader hasn't connected by the time NFC interaction ends, make sure we only
+        // wait for a limited amount of time.
+        if (presentationStateModel.state.value == PresentationStateModel.State.CONNECTING) {
+            val timeout = 15.seconds
```

Comment:
We might want to move this to a constant, s.t. we can easily find and/or reconfigure it. mostly relevant for testing, but still…